### ISSUE TITLE
Direnv Mac eval "$(direnv hook bash)" issue

### ISSUE THREAD

**Issue by billyjacobson** [direnv/direnv#502]
Direnv doesn't seem to want to work with eval "$(direnv hook bash)" when I put it in my .bashrc file. When I open a new tab or terminal window it says it loaded the envrc and then doesn't trigger again. If I manually call eval "$(direnv hook bash)" in the session then it works totally as expected. 

This happens in iterm2 as well as the regular terminal application.

I saw a similar issue that asked about the $PROMPT_COMMAND, so the output of  
echo $PROMPT_COMMAND is 
__bp_precmd_invoke_cmd; __bp_interactive_mode

after I run the hook command though, it is
_direnv_hook;__bp_precmd_invoke_cmd; __bp_interactive_mode

How could I figure out what is resetting it?

**Desktop (please complete the following information):**
 - OS: Mac OS X Mojave
 - Direnv version: 2.20.0

**Comment by kalbasit** [comment]
Are you sure your `.bashrc` is being sourced? This does not happen by the default bash on Mac, and you'll need to source it from your bash_profile

**Comment by billyjacobson** [comment]
Yup, it is being sourced and is the last thing in both files. I also see 
"direnv: loading .envrc"
every time I open a new tab or window, it just doesn't hook into the path changes

**Comment by zimbatm** [comment]
Can you add a `echo "loading direnv"` just before the hook in your .bashrc and confirm that it's printed on every new shell?
Then give me the output of `echo $PROMPT_COMMAND` just after the shell has been loaded just to be sure.

If the "loading direnv" is not printed then the .bashrc has not loaded. If the $PROMPT_COMMAND doesn't contain `_direnv_hook` then something else is messing with the $PROMPT_COMMAND.

It would also be good to get the output of `type __bp_precmd_invoke_cmd` to see what it's doing.

**Comment by billyjacobson** [comment]
I think something got messed up when I installed the iTerm2 shell integration and then uninstalled it. I re-added it and then added the hook underneath the import of it and it is working now. I still have some funky stuff going on, but I don't think it is related to direnv. Thank you

**Comment by billyjacobson** [comment]
Putting `unset PROMPT_COMMAND` at the top of my bash_profile helps with all of this.

**Comment by zimbatm** [comment]
Usually if _direnv_hook is not the last item in $PROMPT_COMMAND is can also cause some issues

**Comment by mirao** [comment]
I've just had a weird issue with direnv hook on Ubuntu 22.04.
The hook was not run from `.bashrc` when I started a terminal and I had to run the hook manually: `eval "$(direnv hook bash)"`.
Finally I fixed it by putting of `eval "$(direnv hook bash)"` after `export PROMPT_COMMAND='history -a'` in my `.bashrc` (originally the hook was placed before the `export PROMPT_COMMAND`).
I don't know why it worked well till today (I'm not aware of any any change in configuration file recently), maybe an update in OS "broke" it ...
